### PR TITLE
Use `Monitor` to resolve `stable_baselines3`'s warning

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

I found one warning message from `stable_baselines3` when I ran `sb3_simple.py` as follows.

```python
$ python sb3_simple.py
[I 2023-01-08 02:56:40,354] A new study created in memory with name: no-name-9ea8a05b-0e13-4c14-8fbc-5339a78a4af8
/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.10/site-packages/stable_baselines3/common/evaluation.py:65: UserWarning: Evaluation environment is not wrapped with a ``Monitor`` wrapper. This may result in reporting modified episode lengths and rewards, if other wrappers happen to modify these. Consider wrapping environment first with ``Monitor`` wrapper.
```

This PR removes the warning message (& fixes a typo, and improves comments by following the coding style of Optuna).

Note that `stable-baselines3`'s version was `1.6.2` when I ran the example code.

## Description of the changes

Wrap gym by `Monitor`.

### CODE REVIEW

File: rl/sb3_simple.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """ Optuna example that optimizes the hyperparameters of
 a reinforcement learning agent using A2C implementation from Stable-Baselines3
-on a OpenAI Gym environment.
+on an OpenAI Gym environment.
```
Comment (toshihikoyanase) [inline rl/sb3_simple.py:3]: Good catch!

File: rl/sb3_simple.py [hunk 2]
```diff
@@ -113,14 +114,13 @@ def _on_step(self) -> bool:
...
     kwargs.update(sample_a2c_params(trial))
-    # Create the RL model
+    # Create the RL model.
     model = A2C(**kwargs)
-    # Create env used for evaluation
-    eval_env = gym.make(ENV_ID)
-    # Create the callback that will periodically evaluate
-    # and report the performance
+    # Create env used for evaluation.
+    eval_env = Monitor(gym.make(ENV_ID))
```
Comment (toshihikoyanase) [inline rl/sb3_simple.py:122]: [Note] `Monitor` inherits `gym.Env`, and we don't need to update the type hints of `TrialEvalCallback`.

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you.

File: rl/sb3_simple.py (hunk 2 above)
Comment (toshihikoyanase) [reply rl/sb3_simple.py:122]: I also confirmed that `Monitor` exists in v0.7.0. So, we can keep the lower version of sb3.
https://github.com/DLR-RM/stable-baselines3/blob/494ebfd20abe90acc136fdaf215c76ec566acd2c/stable_baselines3/common/monitor.py#L15

Comment (nzw0301) [reply rl/sb3_simple.py:122]: Good to know that! Thank you for your investigation!

**Pull request merged**